app-store-connect/src/main.rs

```diff
@@ -0,0 +1,469 @@
+use anyhow::Result;
+use app_store_connect::bundle_api::{BundleId, BundleIdPlatform};
+use app_store_connect::certs_api::{self, Certificate, CertificateType};
+use app_store_connect::device_api::Device;
+use app_store_connect::profile_api::{Profile, ProfileType};
+use app_store_connect::{AppStoreConnectClient, UnifiedApiKey};
+use clap::{Parser, Subcommand};
+use std::io::Write;
+use std::path::PathBuf;
+
+#[derive(Parser)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+    let args = Args::parse();
+    args.command.run()
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Generates a PEM encoded RSA2048 signing key
+    GenerateKey {
+        /// Path to unified api key.
+        #[clap(long)]
+        api_key: PathBuf,
+        /// Certificate type can be one of development, distribution or notarization.
+        #[clap(long)]
+        r#type: String,
```

Comment:
Should this be an enum? Can claps auto derived parser handle that?

Reply:
well yes, but then we'd need to convert the cli enum to the library enum so wouldn't save much in terms of code. I guess error messages would be a bit better

Reply:
I _think_ clap has support for automatically using `FromStr` or something. Pretty sure I've relied on automatic string -> enum conversion before. But can't remember where. Maybe I'm completely wrong about clap's ability to do this.